### PR TITLE
Get schedule by time

### PR DESCRIPTION
**Pull request by RishiVaya**
created an endpoint that gets schedule by satellite id and checks if any given timestamp falls between startDate and endDate, only for schedules for 'FUTURE'.

### CODE REVIEW

File: src/leo-client-app/package.json [hunk 1]
```diff
@@ -22,7 +22,8 @@
     "moment": "^2.29.4",
     "next": "^13.5.4",
     "react": "^18",
-    "react-dom": "^18"
+    "react-dom": "^18",
+    "suncalc": "^1.9.0"
```
Comment (quinnha) [inline src/leo-client-app/package.json:26]: i dont think we need suncalc in the client, does it not work without? we only use suncalc in the server

File: src/leo-client-app/yarn.lock [hunk 2]
```diff
@@ -3046,6 +3046,11 @@ stylis@4.2.0:
   resolved "https://registry.npmjs.org/stylis/-/stylis-4.2.0.tgz"
   integrity sha512-Orov6g6BB1sDfYgzWfTHDOxamtX1bE/zo104Dh9e6fqJ3PooipYyfJ0pUmrZO2wAvO8YbEyeFrkV91XTsGMSrw==
 
+suncalc@^1.9.0:
+  version "1.9.0"
+  resolved "https://registry.npmjs.org/suncalc/-/suncalc-1.9.0.tgz"
+  integrity sha512-vMJ8Byp1uIPoj+wb9c1AdK4jpkSKVAywgHX0lqY7zt6+EWRRC3Z+0Ucfjy/0yxTVO1hwwchZe4uoFNqrIC24+A==
+
```
Comment (quinnha) [inline src/leo-client-app/yarn.lock:3053]: same here

**Review by quinnha — COMMENTED** [review]
try to get it work without using suncalc in the client because im pretty sure we dont use it there (could be wrong)

File: src/leo-client-app/yarn.lock (hunk 2 above)
Comment (RishiVaya) [reply src/leo-client-app/yarn.lock]: removed

File: src/leo-server-app/src/routes/schedule.ts [hunk 3]
```diff
@@ -313,6 +321,33 @@ router.get(
...
+
+    const convertedTime = new Date(time);
+
+    const filter = {
+      satelliteId: satelliteId,
+      status: status,
+      $and: [
+        {startDate: { '$lte': convertedTime }} ,
+        { endDate: { '$gte': convertedTime } },
+      ],
```
Comment (Umang-Rajkarnikar) [inline src/leo-server-app/src/routes/schedule.ts:341]: May be wrong but shouldn't this be switched?

**Review by quinnha — APPROVED** [review]
lgtm

File: src/leo-server-app/src/routes/schedule.ts (hunk 3 above)
Comment (RishiVaya) [reply src/leo-server-app/src/routes/schedule.ts:341]: it says if start date is less than equal to input time or end date greeater or eqaul to input time

**Pull request merged**